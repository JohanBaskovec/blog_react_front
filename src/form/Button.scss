.Button {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0.6rem 1rem;
  background-color: #4089d7;
  color: white;
  border: none;
  transition: all 0.3s;
  cursor: pointer;

  &:focus {
    outline: 0;
    box-shadow: 0 0 0 .15rem rgba(70, 158, 246, 0.5);
  }
  &:hover {
    background-color: #356caf;
  }
}

.Button--primary {

}
