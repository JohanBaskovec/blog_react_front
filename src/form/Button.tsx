import React from "react";
import "./Button.scss";

export interface ButtonProps {
    children?: React.ReactNode;
    style?: React.CSSProperties;
    onClick?: React.MouseEventHandler<HTMLButtonElement>;
}

export function Button(props: ButtonProps) {
    return (
        <button type="button"
                className="Button"
                onClick={props.onClick}>
            {props.children}
        </button>
    )
}
