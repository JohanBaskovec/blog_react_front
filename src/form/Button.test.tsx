import React from "react";
import TestRenderer from 'react-test-renderer';
import {Button} from "./Button";
import {fireEvent, render, screen} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

describe('Button', () => {
    it("can be clicked", () => {
        let called = false;
        const onClick = () => {
            called = true;
        }
        const element = <Button onClick={onClick}>Submit</Button>;
        render(element);
        expect(screen.getByRole('button', {name: 'Submit'})).not.toBeNull();
        fireEvent.click(screen.getByText('Submit'));
        expect(called).toBe(true);
    });

    it("is rendered properly", () => {
        const onClick = () => {
        }
        const tree = TestRenderer
            .create(<Button onClick={onClick}>Submit</Button>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
