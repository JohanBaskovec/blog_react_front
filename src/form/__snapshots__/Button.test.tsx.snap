// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button is rendered properly 1`] = `
<button
  className="Button"
  onClick={[Function]}
  type="button"
>
  Submit
</button>
`;
